modules/e2eping/e2eping.js

```diff
@@ -0,0 +1,333 @@
+                JitsiConferenceEvents.USER_LEFT,
+                (id, participant) => this.participantLeft(participant));
+        }
+
+        // Listen to E2E PING requests and responses from other participants
+        // in the conference.
+        conference.on(
+            JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+            (participant, payload) => {
+                if (payload.type === E2E_PING_REQUEST) {
+                    this.handleRequest(participant, payload);
+                } else if (payload.type === E2E_PING_RESPONSE) {
+                    this.handleResponse(participant, payload);
+                }
+            });
+
+        // We don't want to wait the whole interval before sending the first
+        // request, but we can't send it immediately after the participant joins
+        // either, because our data channel might not have initialized.
+        // So once the data channel initializes, send requests to everyone.
+        // Wait an additional 200ms to give a chance to the remote side (if it
+        // also just connected as is the case for the first 2 participants in a
+        // conference) to open its data channel.
+        conference.on(
+            JitsiConferenceEvents.DATA_CHANNEL_OPENED,
+            () => {
+                this.dataChannelOpened = true;
+                for (const id in this.participants) {
+                    if (this.participants.hasOwnProperty(id)) {
+                        const participantState = this.participants[id];
+
+                        window.setTimeout(participantState.sendRequest, 200);
+                    }
+                }
+            });
+    }
+
+    /**
+     * Handles a participant joining the conference. Starts to send ping
+     * requests to the participant.
+     * @param {JitsiParticipant} participant the participant that joined.
+     */
+    participantJoined(participant) {
+        const id = participant.getId();
+
+        if (this.participants[id]) {
+            // State state left?
+            logger.info(
+                `Participant state already exists for ${id}. Clearing.`);
+            this.participants[id].clearIntervals();
+            delete this.participants[id];
+        }
+
+        this.participants[id] = new ParticipantState(participant, this);
+    }
+
+    /**
+     * Handles a participant leaving the conference. Stops sending requests.
+     * @param {JitsiParticipant} participant the participant that left.
+     */
+    participantLeft(participant) {
+        const id = participant.getId();
+
+        if (this.participants[id]) {
+            this.participants[id].clearIntervals();
+            delete this.participants[id];
+        }
+    }
+
+    /**
+     * Handles a ping request coming from another participant.
+     * @param {JitsiParticipant} participant the participant who sent the
+     * request.
+     * @param {Object} request the request.
+     */
+    handleRequest(participant, request) {
```

Comment:
if we only use the participant id here, what about just passing the id in directly instead of the participant?

Reply:
This results in more inline code in the constructor

Reply:
changing `this.handleRequest(participant, payload);` to `this.handleRequest(participant.getId(), payload);`?

Reply:
I switched it